import "cell.proto";

package Filesystem;

message Leak {
		required Coordinate location = 1;
		required double concentration = 2;
	}

message Hypothesis {
	required double probability = 1;
	repeated Leak leak = 2;
	repeated Cells methene_cells = 3;
}

message Hypotheses {
	repeated Hypothesis hyp = 1;
	repeated Leak detection = 2;
}

message Hypotheses_history {
    repeated Hypotheses hyps = 1;
}